validator/delphi_validator/validate.py

```diff
@@ -579,18 +579,149 @@ def check_rapid_change_num_rows(self, df_to_test, df_to_reference, checking_date
+            #print(vals_modified_f)
+            group['fmean'] = fmean
+            group['fmedian'] = fmedian
+            group['smedian'] = smedian
+            group['fsd'] = fsd
+            group['ssd'] = ssd
+            group['fmad'] = fmad
+            group['smad'] = smad
+            group['ftstat'] = ftstat
+            group['ststat'] = ststat 
+            all_full_frames.append(group) 
+
+        all_frames = pd.concat(all_full_frames)
+
+        # Determine outliers
+        outlier_source_df = all_frames.sort_values(by=['time_value']).copy()
+        outlier_source_df["flag"] = outlier_source_df.apply(outlier_flag, axis = 1)
+        outlier_group = outlier_source_df.groupby("geo_id")
+
+        outlier_append = pd.DataFrame()
+        for cat, group in outlier_group:
+            group = group.reset_index()
+            for index, row in group.iterrows():
+                if row["flag"] == 1:
+                    try: 
+                        eval_next = group.iloc[index+1, :]
+                        if (not pd.isna(eval_next['ststat'])) and (eval_next['ststat'] > sig_consec):
+                            eval_next["flag"] == 1
+                            outlier_append = outlier_append.append(eval_next, ignore_index=True)
+
+                        if pd.isna(eval_next['ststat']) and  (eval_next['ftstat'] > sig_consec):
+                            eval_next["flag"] == 1
+                            outlier_append = outlier_append.append(eval_next, ignore_index=True)
+
+                    except: 
+                        continue
+                    try: 
+                        eval_prev = group.iloc[index-1, :]
+                        if (not pd.isna(eval_prev['ststat'])) and (eval_prev['ststat'] > sig_consec):
+                            eval_prev["flag"] == 1
+                            outlier_append = outlier_append.append(eval_prev, ignore_index=True)
+                        if pd.isna(eval_prev['ststat']) and  (eval_prev['ftstat'] > sig_consec):
+                            eval_prev["flag"] == 1
+                            outlier_append = outlier_append.append(eval_prev, ignore_index=True)
+                    except: 
+                        continue                 
+
+        outlier_append["flag"] = 1
+        outliers = outlier_source_df[outlier_source_df["flag"] == 1]
+        all_o = pd.concat([outliers, outlier_append]).drop(columns=['index']).sort_values(by=['time_value','geo_id']).drop_duplicates()
+        all_o = all_o.reset_index().drop(columns=['index'])
+        all_o.to_csv(str(checking_date.date()) + sig + "outliers.csv")
+        all_frames.to_csv(str(checking_date.date()) + sig + "all_frames.csv")
+
+        if outliers.shape[0] > 0:
+            self.raised_errors.append(ValidationError(
+                ("data_corrections_range",
+                 (checking_date.date()-timedelta(days=1), checking_date.date()+timedelta(days=1)), geo, sig),
```

Comment:
This was a bit tricky because when you are checking one date (10/25), this method calculates the range of outliers from 10/24-10/26 as available and returns that in the error expression. Any suggestions on what I should do instead?

Reply:
It sounds like this date range is unique for a given checking_date, so just the checking_date should be sufficient in this first arg. You can put the checked date range into the actual error message, e.g., "found outliers in date range 10/24-10/26 compared to last 30 days of data" to provide more info about the result.

If I'm not understanding something, we can set up a time to talk.